nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -300,7 +312,11 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
             if (isInactive) {
                 final boolean continual = context.getProperty(CONTINUALLY_SEND_MESSAGES).asBoolean();
-                sendInactiveMarker = !inactive.getAndSet(true) || (continual && (now > lastInactiveMessage.get() + thresholdMillis));
+                if (!initialInactivityIndicator) {
+                    sendInactiveMarker = (!inactive.getAndSet(true) || (continual && (now > lastInactiveMessage.get() + thresholdMillis))) && hasSuccessTransfer.get();
+                } else {
+                    sendInactiveMarker = !inactive.getAndSet(true) || (continual && (now > lastInactiveMessage.get() + thresholdMillis));
+                }
```

Comment:
`!inactive.getAndSet(true) || (continual && (now > lastInactiveMessage.get()` is common code between lines 316 and 318. Can you please assign this to a variable whose name should describe what these conditions are and then use the variable on lines 316 and 318? Thanks!